cli/proc_state.rs

```diff
@@ -574,23 +575,21 @@ impl ProcState {
           | MediaType::Json => code.as_ref().clone(),
           MediaType::Dts => "".to_string(),
           _ => {
-            let emit_path = self
+            match self
               .dir
               .gen_cache
-              .get_cache_filename_with_extension(&found, "js")
```

Comment:
We had multiple ways for getting this data from the cache. Here it was accessing it via `DiskCache` and other places it was `FetchCache`. I've combined them to one api (`.get_emit_data`)